tests/system-tests/rdscore/internal/rdscorecommon/egress-ip.go

```diff
@@ -0,0 +1,705 @@
+		var parsedIP string
+
+		err := wait.PollUntilContextTimeout(
+			context.TODO(),
+			time.Second,
+			time.Second*5,
+			true,
+			func(ctx context.Context) (bool, error) {
+				result, err := clientPod.ExecCommand(cmdToRun, clientPod.Object.Spec.Containers[0].Name)
+
+				if err != nil {
+					glog.V(100).Infof("Error running command from within a pod %q: %v",
+						clientPod.Object.Name, err)
+
+					return false, nil
+				}
+
+				glog.V(100).Infof("Successfully executed command from within a pod %q: %v",
+					clientPod.Object.Name, err)
+				glog.V(100).Infof("Command's output:\n\t%v", result.String())
+
+				parsedIP, _, err = net.SplitHostPort(result.String())
+
+				if err != nil {
+					glog.V(100).Infof("Failed to parse %q for host/port pair", result.String())
+
+					return false, nil
+				}
+
+				glog.V(100).Infof("Verify IP version type correctness")
+
+				myIP, err := netip.ParseAddr(parsedIP)
+
+				if err != nil {
+					glog.V(100).Infof("Failed to parse used ip address %q", parsedIP)
+
+					return false, nil
+				}
+
+				if isIPv6 && myIP.Is4() || !isIPv6 && myIP.Is6() {
+					glog.V(100).Infof("Wrong IP version detected; %q", parsedIP)
+
+					return false, nil
+				}
+
+				glog.V(100).Infof("Comparing %q with expected %q", parsedIP, expectedIPs)
+
+				for _, expectedIP := range expectedIPs {
+					if parsedIP == expectedIP {
+						return true, nil
+					}
+				}
+
+				glog.V(100).Infof("Mismatched IP address. Expected %q got %q", expectedIPs, parsedIP)
+
+				return false, nil
+			})
+
+		if err != nil {
+			return fmt.Errorf("failed to run command from within pod %s: %w", clientPod.Object.Name, err)
+		}
+	}
+
+	return nil
+}
+
+func getExpectedEgressIPList() ([]string, error) {
+	By("Getting a map of source nodes and assigned Egress IPs for these nodes")
+
+	egressIPObj, err := egressip.Pull(APIClient, RDSCoreConfig.EgressIPName)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to pull egressIP %q object: %v", RDSCoreConfig.EgressIPName, err)
+
+		return nil, fmt.Errorf("failed to retrieve egressIP %s object: %w", RDSCoreConfig.EgressIPName, err)
+	}
+
+	egressIPSet, err := egressIPObj.GetAssignedEgressIPMap()
+
+	if err != nil {
+		glog.V(100).Infof("Failed to retrieve egressIP %s assigned egressIPs map: %v",
+			RDSCoreConfig.EgressIPName, err)
+
+		return nil, fmt.Errorf("failed to retrieve egressIP %s assigned egressIPs map: %w",
+			RDSCoreConfig.EgressIPName, err)
+	}
+
+	if len(egressIPSet) != 2 {
+		glog.V(100).Infof("EgressIPs assigned to the wrong number of nodes: %v", egressIPSet)
+
+		return nil, fmt.Errorf("egressIPs assigned to the wrong number of nodes: %v", egressIPSet)
+	}
+
+	for _, nodeName := range []string{RDSCoreConfig.EgressIPNodeOne, RDSCoreConfig.EgressIPNodeTwo} {
+		egressIPOne := egressIPSet[RDSCoreConfig.EgressIPNodeOne]
+		egressIPTwo := egressIPSet[RDSCoreConfig.EgressIPNodeTwo]
+
+		if egressIPOne != RDSCoreConfig.EgressIPv4 || egressIPTwo != RDSCoreConfig.EgressIPv6 {
```

Comment:
is there a way we can enforce `RDSCoreConfig.EgressIPNodeOne` to always have `RDSCoreConfig.EgressIPv4`?

Reply:
it depends (on the assigned nodes count as well), but in generally - no; anyway this code block was change; please take a look on the latest version